hal_espressif expects ZEPHYR_BASE environment variable to be set

ESP32 targets do not build properly when the `ZEPHYR_BASE` environment variable is not set (i.e. when the Zephyr CMake package from the registry is used to find the Zephyr path).

```
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/devel/Dev/zephyrproject/zephyr/tests/kernel/threads/thread_apis/build/esp-idf/build/sdkconfig
Traceback (most recent call last):
  File "/home/devel/Dev/zephyrproject/modules/hal/espressif/tools/kconfig_new/confgen.py", line 42, in <module>
    sys.path.insert(0, os.path.join(os.environ['ZEPHYR_BASE'], "scripts", "kconfig"))
  File "/usr/lib/python3.8/os.py", line 675, in __getitem__
    raise KeyError(key) from None
KeyError: 'ZEPHYR_BASE'
CMake Error at /home/devel/Dev/zephyrproject/modules/hal/espressif/tools/cmake/kconfig.cmake:228 (message):
  Failed to run confgen.py
  (/usr/bin/python3.8;/home/devel/Dev/zephyrproject/modules/hal/espressif/tools/kconfig_new/confgen.py;--kconfig;/home/devel/Dev/zephyrproject/modules/hal/espressif/Kconfig;--sdkconfig-rename;/home/devel/Dev/zephyrproject/modules/hal/espressif/sdkconfig.rename;--config;/home/devel/Dev/zephyrproject/zephyr/tests/kernel/threads/thread_apis/build/esp-idf/build/sdkconfig;--env-file;/home/devel/Dev/zephyrproject/zephyr/tests/kernel/threads/thread_apis/build/esp-idf/build/bootloader/config.env).
  Error 1
Call Stack (most recent call first):
  /home/devel/Dev/zephyrproject/modules/hal/espressif/tools/cmake/build.cmake:451 (__kconfig_generate_config)
  /home/devel/Dev/zephyrproject/modules/hal/espressif/tools/cmake/project.cmake:396 (idf_build_process)
  CMakeLists.txt:41 (project)
```

The `ZEPHYR_BASE` should be sourced from the `ZEPHYR_BASE` CMake variable as done here:
https://github.com/zephyrproject-rtos/zephyr/blob/70a1827fc1cbe65f5a7b8217f8d6259680d3974f/cmake/modules/kconfig.cmake#L156-L157

--- a/scripts/kconfig/kconfigfunctions.py
+++ b/scripts/kconfig/kconfigfunctions.py
@@ -6,8 +6,9 @@
 import os
 import pickle
 import sys
+from pathlib import Path
 
-ZEPHYR_BASE = os.environ["ZEPHYR_BASE"]
+ZEPHYR_BASE = str(Path(__file__).resolve().parents[2])
 sys.path.insert(0, os.path.join(ZEPHYR_BASE, "scripts", "dts",
                                 "python-devicetree", "src"))
 

--- a/west.yml
+++ b/west.yml
@@ -65,7 +65,7 @@ manifest:
       groups:
         - hal
     - name: hal_espressif
-      revision: 12d88df9caf69c8b4f3ad44efc7780b0f0e50db2
+      revision: bceaadfc8f4f812ace778937a6b042e3b6249215
       path: modules/hal/espressif
       west-commands: west/west-commands.yml
       groups:

